### PR TITLE
feat(launcher): Forward JobResources.accelerators/cpus/memory to sbatch

### PR DESCRIPTION
**Pull request by min-oumi**
## Summary

`SlurmCluster.run_job` currently logs "Accelerators are unused for Slurm jobs" and drops `JobResources.accelerators`, `cpus`, and `memory` on the floor. Any job submitted with `cloud="slurm"` and `accelerators="H100:8"` runs as CPU-only and fails on GPU-dependent code. This forwards them to `sbatch` as `--gres` / `--cpus-per-task` / `--mem`.

## Changes

- `_parse_accelerators_to_gres()` — translates oumi accelerator strings into Slurm `--gres` values

  | Input | Output |
  |---|---|
  | `"H100:8"` | `"gpu:H100:8"` |
  | `"H100"` | `"gpu:H100:1"` |
  | `"A100-80GB:4"` | `"gpu:A100-80GB:4"` |
  | `":8"` | `"gpu:8"` (untyped) |
  | `"8"` | `"gpu:8"` (untyped) |
  | `"H100:8+"` (SkyPilot "at least") | `"gpu:H100:8"` |
  | `"8+"` | `"gpu:8"` |

- `_resources_to_sbatch_kwargs()` — collects gres + cpus_per_task + mem from a `JobResources`. Strips SkyPilot's `"+"` minimum-modifier across all three fields. Preserves explicit memory unit suffixes (`32G`, `2T`); defaults to `G` for bare numbers (per `JobResources.memory` doc: "in GiB").

- `SlurmCluster.run_job` forwards the kwargs to `SlurmClient.submit_job`, which already routes unknown kwargs to `--<flag>=<value>` on the sbatch command.

- `_validate_job_config` no longer warns for fields we now forward (accelerators, cpus, memory). Region/zone/instance_type/disk_size warnings remain — those still don't have a Slurm mapping.

## Tests

- 13 new unit tests covering the helper edge cases, typed vs untyped GRES, SkyPilot modifier stripping (incl. `H100:8+` and `8+`), explicit memory units, and the bare-resources case (cluster defaults take effect when nothing's pinned).
- 7 existing `run_job` assertions updated to expect the new kwargs (the default test fixture uses `accelerators="A100-80GB" cpus="4" memory="64"` — those now propagate).
- All 82 slurm cluster/client tests pass.

Skipped local pyright (`SKIP=pyright`) — pre-existing `src/oumi/cli/analyze.py` errors on `origin/main` unrelated to this change.

### Direct `oumi.launcher.up` (standalone)

```python
oumi.launcher.up(
    JobConfig(
        cloud="slurm",
        resources=JobResources(cloud="slurm", accelerators="H100:1"),
        run="echo CUDA_VISIBLE_DEVICES=$CUDA_VISIBLE_DEVICES; nvidia-smi -L",
    ),
    cluster_name="root@<controller>",
)
```

Job 214 on RunPod cluster: `JobState=COMPLETED ExitCode=0:0`, `TresPerNode=gres:gpu:H100:1`, stdout shows `CUDA_VISIBLE_DEVICES=6` + `nvidia-smi -L` listing the H100s.

### Full worker stack (curl → API → workflow → run_job_activity → oumi → sbatch)

Submitted a training request via the local API (`POST /v1/projects/{id}/models:train`, `cloud_provider="slurm"`, `accelerator="H100:1"` via `TRAINING_ACCELERATOR_TIERS` override). Worker dispatched Slurm job 219:

```
JobId=219 JobName=job-172-1317
   JobState=FAILED Reason=NonZeroExitCode ExitCode=127:0   ← script needed `sudo apt-get`, unrelated
   BatchHost=node-1
   TRES=cpu=2,node=1,billing=2
   TresPerNode=gres:gpu:H100:1                              ← --gres correctly forwarded
```

Confirmed via a temporary `logger.info` (now removed) that the worker's call into oumi was:
```
_resources_to_sbatch_kwargs: accelerators='H100:1' cpus=None memory=None -> kwargs={'gres': 'gpu:H100:1'}
```

So the resource forwarding correctly traverses: API request → worker JobConfig → oumi `SlurmCluster.run_job` → `SlurmClient.submit_job` → `sbatch --gres=gpu:H100:1`.

Before this PR: the same path produced `TresPerNode=` empty (the warning fired, accelerators dropped).

## Caveat for cluster operators

Typed-GRES requires the type string to match an entry in the cluster's `gres.conf`. If the user passes `"H100:8"` and the cluster only has untyped GRES (`gres.conf` doesn't define `Type=H100`), sbatch will reject the request. Workaround: pass `":8"` or `"8"` to skip the type, or fix `gres.conf` upstream.

### CODE REVIEW

File: src/oumi/launcher/clusters/slurm_cluster.py [hunk 1]
```diff
@@ -98,6 +99,71 @@ def _create_job_script(job: JobConfig) -> str:
...
+    Examples:
+        ``"H100:8"``      -> ``"gpu:H100:8"``
+        ``"H100"``        -> ``"gpu:H100:1"``
+        ``"A100-80GB:4"`` -> ``"gpu:A100-80GB:4"``
+        ``":8"``          -> ``"gpu:8"`` (untyped count)
+        ``"8"``           -> ``"gpu:8"`` (untyped count)
+    """
+    if not accelerators:
+        return None
+    spec = accelerators.strip()
```
[the remark this answers is not in the document]
Comment (min-oumi) [reply src/oumi/launcher/clusters/slurm_cluster.py]: Good catch — both cases are now handled in d40ac5851:

- `accelerators` runs through `_strip_modifier` before parsing → `"H100:8+"` → `"gpu:H100:8"`, `"8+"` → `"gpu:8"`
- Count post-partition gets stripped too (handles `"H100:8+"` where the `+` is after the colon)
- Also fixed a related ordering bug in `_strip_modifier` — trailing whitespace before `+` (e.g. `" 8+ "`) used to prevent stripping; now strips whitespace first

5 new parametrize cases: `"H100:8+"`, `"8+"`, `" H100:8+ "`, `"   "`, `"+"`.

[the remark this answers is not in the document]
Comment (min-oumi) [reply src/oumi/launcher/clusters/slurm_cluster.py]: Good point, fixing

File: src/oumi/launcher/clusters/slurm_cluster.py [hunk 2]
```diff
@@ -115,26 +203,19 @@ def _validate_job_config(job: JobConfig) -> None:
             f"`Resources.cloud` must be `slurm`. "
             f"Unsupported cloud: {job.resources.cloud}"
         )
-    # Warn that other resource parameters are unused for Slurm.
     if not job.working_dir:
         logger.warning("Working directory is not set. This is not recommended.")
+    # Resource fields that don't map to sbatch flags — warn so users know
```
Comment (jgreer013) [inline src/oumi/launcher/clusters/slurm_cluster.py:208]: Run `/pr-review` skill here pls :)

**Review by jgreer013 — APPROVED** [review]
